base/src/main/java/org/mozilla/jss/pkcs11/PK11KeyWrapper.java

```diff
@@ -291,29 +291,37 @@ private AlgorithmParameterSpec checkParams(AlgorithmParameterSpec params)
 
         assert( pubKey!=null && privKey==null && symKey==null );
         NativeProxy params = null;
-        long params_size = 0;
+        long paramsSize = 0;
         if (parameters != null) {
             try {
                 ((NativeEnclosure) parameters).open();
                 params = ((NativeEnclosure) parameters).mPointer;
-                params_size = ((NativeEnclosure) parameters).mPointerSize;
+                paramsSize = ((NativeEnclosure) parameters).mPointerSize;
             } catch (Exception e) {
                 throw new TokenException(e.getMessage(), e);
             }
         }
-
+        byte[] sk = null;
+        TokenException te = null;
         try {
-            return nativeWrapSymWithPub(token, toBeWrapped, pubKey,
-                                        algorithm, params, params_size);
-        } finally {
-            if (parameters != null) {
-                try {
-                    ((NativeEnclosure) parameters).close();
-                } catch (Exception e) {
-                    throw new TokenException(e.getMessage(), e);
+            sk = nativeWrapSymWithPub(token, toBeWrapped, pubKey,
+                                        algorithm, params, paramsSize);
+        } catch (TokenException e) {
+            te = e;
+        }
+        if (parameters != null) {
+            try {
+                ((NativeEnclosure) parameters).close();
+            } catch (Exception e) {
+                if (te == null) {
+                    te =  new TokenException(e.getMessage(), e);
                 }
```

Comment:
If `te` is not  `null`, the `e` will be buried instead, and we will never know if something went wrong in `NativeEnclosure.close()` which might cause a memory leak. I'd suggest to log an error message with the stack trace since the `NativeEnclosure.close()` is not supposed to fail.